doc/Data Plane Code Generation/Subclients.md

```diff
@@ -0,0 +1,247 @@
+        /// <exception cref="ArgumentNullException"> <paramref name="credential"/> is null. </exception>
+        public SubClientsClient(AzureKeyCredential credential, Uri endpoint = null, SubClientsClientOptions options = null)
+        {
+            Argument.AssertNotNull(credential, nameof(credential));
+            endpoint ??= new Uri("http://localhost:3000");
+            options ??= new SubClientsClientOptions();
+
+            ClientDiagnostics = new ClientDiagnostics(options);
+            _keyCredential = credential;
+            _pipeline = HttpPipelineBuilder.Build(options, Array.Empty<HttpPipelinePolicy>(), new HttpPipelinePolicy[] { new AzureKeyCredentialPolicy(_keyCredential, AuthorizationHeader) }, new ResponseClassifier());
+            _endpoint = endpoint;
+        }
+
+        private Parameter _cachedParameter;
+
+        /// <summary> Initializes a new instance of Root. </summary>
+        /// <param name="cachedParameter"> The String to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="cachedParameter"/> is null. </exception>
+        /// <exception cref="ArgumentException"> <paramref name="cachedParameter"/> is an empty string, and was expected to be non-empty. </exception>
+        public virtual Root GetRootClient(string cachedParameter)
+        {
+            Argument.AssertNotNullOrEmpty(cachedParameter, nameof(cachedParameter));
+
+            return new Root(ClientDiagnostics, _pipeline, _keyCredential, cachedParameter, _endpoint);
+        }
+
+        /// <summary> Initializes a new instance of Parameter. </summary>
+        public virtual Parameter GetParameterClient()
+        {
+            return Volatile.Read(ref _cachedParameter) ?? Interlocked.CompareExchange(ref _cachedParameter, new Parameter(ClientDiagnostics, _pipeline, _keyCredential, _endpoint), null) ?? _cachedParameter;
+        }
+    }
+}
+
+// SubClient: Root Generated\Root.cs
+namespace Azure.Service.SubClients
+{
+-   public partial class RootClient
++   public partial class Root
+    {
+-       protected RootClient(){}
++       protected Root(){}
+-       public RootClient(string cachedParameter, AzureKeyCredential credential, Uri endpoint = null, SubClientsClientOptions options = null){}
++       internal Root(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, AzureKeyCredential keyCredential, string cachedParameter, Uri endpoint){}
+    }
+}
+
+//subclient: Parameter Generated\Parameter.cs
+namespace Azure.Service.SubClients
+{
+-   public partial class ParameterClient
++   public partial class Parameter
+    {
+-       protected ParameterClient()
++       protected Parameter()
+-       public ParameterClient(AzureKeyCredential credential, Uri endpoint = null, SubClientsClientOptions options = null){}
++       internal Parameter(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, AzureKeyCredential keyCredential, Uri endpoint) {}
+    }
+}
+
+```
+
+</details>
+
+### Client hierarchy
```

Comment:
This makes Root the single top level client.  Why would I use one approach vs. another?

Reply:
Use 'single top-level client' approach, it will use the service client (SubClientsClient) as the single top-level client, other clients are its sub-clients.  
Use 'Client hierarchy' approach, you can select one client as the top-level client, others are sub-clients according to the client hierarchy defined in customization.cs